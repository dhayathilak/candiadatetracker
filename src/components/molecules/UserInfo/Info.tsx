
import { Button, Card, CardContent, Stack,Grid} from "@mui/material"
import { makeStyles } from "@mui/styles"
import { string } from "prop-types"
import { ReactElement } from "react"
import UserIcon from "../../atoms/Avatar/userIcon"
import ButtonComponent from "../../atoms/Button/ButtonComponent"
import Text from "../../atoms/Typography/Text"




interface propTypes{
    header: string,
    detail: string,
    icontype: ReactElement,
    name?:any
}

const useStyles = makeStyles({
    icons:{
     boxSizing: 'border-box',
     border: '1px solid #E5E7ED',
     width:'44px',
     height:'44px',
     color: '#95AAFF',
  
     borderRadius:'12px'
    }
  })

const Info = (props: propTypes)=>{
    const classes = useStyles()
    return(
        <>
                    <Card>
                        <CardContent >
                                <Stack direction='row' spacing={1}>
                                    <ButtonComponent icon={props.icontype} type={classes.icons}/>
                                    <Stack direction='column'>
                                        <Text text={props.header}/>
                                        <Text text={props.detail}/>
                                    </Stack>
                                </Stack> 
                        </CardContent>
                    </Card>
        </>
      
    )
}

export default Info