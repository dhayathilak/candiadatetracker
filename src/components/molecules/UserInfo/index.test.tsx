import { Person2 } from "@mui/icons-material"
import { render } from "@testing-library/react"
import Info from "."

test('userinfo',()=>{
    expect(render(<Info header='head' detail="mydetail" icontype={<Person2/>}/>)).toBeDefined()
    expect(render(<Info header='head' detail="mydetail" icontype={<Person2/>}/>)).toMatchSnapshot()

})