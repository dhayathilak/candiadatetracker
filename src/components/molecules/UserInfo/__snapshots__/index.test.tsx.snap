// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userinfo 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="css-w4z10b-MuiStack-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-makeStyles-icons-1 MuiButton-makeStyles-icons-1Primary MuiButton-sizeMedium MuiButton-makeStyles-icons-1SizeMedium MuiButton-root MuiButton-makeStyles-icons-1 MuiButton-makeStyles-icons-1Primary MuiButton-sizeMedium MuiButton-makeStyles-icons-1SizeMedium css-1w1rijm-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="Person2Icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18.39 14.56C16.71 13.7 14.53 13 12 13s-4.71.7-6.39 1.56C4.61 15.07 4 16.1 4 17.22V20h16v-2.78c0-1.12-.61-2.15-1.61-2.66zM9.78 12h4.44c1.21 0 2.14-1.06 1.98-2.26l-.32-2.45C15.57 5.39 13.92 4 12 4S8.43 5.39 8.12 7.29L7.8 9.74c-.16 1.2.77 2.26 1.98 2.26z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <div
              class="css-nen11g-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                head
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                mydetail
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="css-w4z10b-MuiStack-root"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-makeStyles-icons-1 MuiButton-makeStyles-icons-1Primary MuiButton-sizeMedium MuiButton-makeStyles-icons-1SizeMedium MuiButton-root MuiButton-makeStyles-icons-1 MuiButton-makeStyles-icons-1Primary MuiButton-sizeMedium MuiButton-makeStyles-icons-1SizeMedium css-1w1rijm-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="Person2Icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18.39 14.56C16.71 13.7 14.53 13 12 13s-4.71.7-6.39 1.56C4.61 15.07 4 16.1 4 17.22V20h16v-2.78c0-1.12-.61-2.15-1.61-2.66zM9.78 12h4.44c1.21 0 2.14-1.06 1.98-2.26l-.32-2.45C15.57 5.39 13.92 4 12 4S8.43 5.39 8.12 7.29L7.8 9.74c-.16 1.2.77 2.26 1.98 2.26z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <div
              class="css-nen11g-MuiStack-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                head
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                mydetail
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          class="css-w4z10b-MuiStack-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-makeStyles-icons-1 MuiButton-makeStyles-icons-1Primary MuiButton-sizeMedium MuiButton-makeStyles-icons-1SizeMedium MuiButton-root MuiButton-makeStyles-icons-1 MuiButton-makeStyles-icons-1Primary MuiButton-sizeMedium MuiButton-makeStyles-icons-1SizeMedium css-1w1rijm-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="Person2Icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18.39 14.56C16.71 13.7 14.53 13 12 13s-4.71.7-6.39 1.56C4.61 15.07 4 16.1 4 17.22V20h16v-2.78c0-1.12-.61-2.15-1.61-2.66zM9.78 12h4.44c1.21 0 2.14-1.06 1.98-2.26l-.32-2.45C15.57 5.39 13.92 4 12 4S8.43 5.39 8.12 7.29L7.8 9.74c-.16 1.2.77 2.26 1.98 2.26z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="css-nen11g-MuiStack-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              head
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
            >
              mydetail
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
